src/bthome_ble/parser.py

```diff
@@ -295,48 +295,77 @@ def _parse_payload(self, payload: bytes) -> bool:
         payload_length = len(payload)
         next_obj_start = 0
         result = False
+        measurements: list[dict[str, Any]] = []
+        device_id_dict: dict[int, int] = {}
 
+        # Create a list with all individual objects
         while payload_length >= next_obj_start + 1:
             obj_start = next_obj_start
-
             obj_control_byte = payload[obj_start]
-            obj_meas_type = payload[obj_start + 1]
             obj_data_length = (obj_control_byte >> 0) & 31  # 5 bits (0-4)
             obj_data_format = (obj_control_byte >> 5) & 7  # 3 bits (5-7)
+            obj_meas_type = payload[obj_start + 1]
 
-            obj_end = obj_start + obj_data_length
-            next_obj_start = obj_end + 1
+            if obj_data_length == 0:
```

Comment:
is it legal to be 0?

Reply:
If it's not, we should log a debug log that we skipped invalid data.

Reply:
someone can accidently set the length byte to 0, so log a debug message sounds like the right way. If the next byte is a new measurement, it should parse the rest of the message, but most likely, the entire message will incorrect probably. I added a log message